src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,117 @@
+package duke.tasks;
+
+/**
+ * Represents a generic task with a specific type, description, and completion status.
+ *
+ * <p> The Task class is a fundamental representation of a task that can be of various types
+ * such as TODO, DEADLINE, or EVENT.
+ * Each task has a type, a description, and a flag that indicates whether it has been completed. </p>
+ */
+public class Task {
+    private TaskType type;
+    private String description;
+    private boolean done;
```

Comment:
Perhaps we can rename this to `isDone` to make it even clearer that the variable name is a boolean?